Intermittent CI failures with `resolvelib`

These failures seem to occur intermittently on all unit tests that use `resolvelib`:

```
____________________ test_requirement_source_multiple_files ____________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fccbeeebb80>

    def test_requirement_source_multiple_files(monkeypatch):
        file1 = "requirements1.txt"
        file2 = "requirements2.txt"
        file3 = "requirements3.txt"
    
        source = requirement.RequirementSource(
            [Path(file1), Path(file2), Path(file3)],
            ResolveLibResolver(),
        )
    
        def read_file_mock(f):
            filename = f.name
            if filename == file1:
                return ["flask==2.0.1"]
            elif filename == file2:
                return ["requests==2.8.1"]
            else:
                assert filename == file3
                return ["pip-api==0.0.22\n", "packaging==21.0"]
    
        monkeypatch.setattr(_parse_requirements, "_read_file", read_file_mock)
    
>       specs = list(source.collect())

test/dependency_source/test_requirement.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pip_audit/dependency_source/requirement.py:33: in collect
    for _, deps in self.resolver.resolve_all(iter(req_values)):
pip_audit/dependency_source/interface.py:55: in resolve_all
    yield (req, self.resolve(req))
pip_audit/dependency_source/resolvelib/resolvelib.py:27: in resolve
    result = self.resolver.resolve([req])
env/lib/python3.8/site-packages/resolvelib/resolvers.py:482: in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
env/lib/python3.8/site-packages/resolvelib/resolvers.py:373: in resolve
    name = min(unsatisfied_names, key=self._get_preference)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <resolvelib.resolvers.Resolution object at 0x7fccbee6bf40>
name = 'flask'

    def _get_preference(self, name):
>       return self._p.get_preference(
            identifier=name,
            resolutions=self.state.mapping,
            candidates=IteratorMapping(
                self.state.criteria,
                operator.attrgetter("candidates"),
            ),
            information=IteratorMapping(
                self.state.criteria,
                operator.attrgetter("information"),
            ),
            backtrack_causes=self.state.backtrack_causes,
        )
E       TypeError: get_preference() got an unexpected keyword argument 'backtrack_causes'

env/lib/python3.8/site-packages/resolvelib/resolvers.py:178: TypeError
```

--- a/pip_audit/dependency_source/resolvelib/pypi_provider.py
+++ b/pip_audit/dependency_source/resolvelib/pypi_provider.py
@@ -161,7 +161,7 @@ class PyPIProvider(AbstractProvider):
     def identify(self, requirement_or_candidate):
         return canonicalize_name(requirement_or_candidate.name)
 
-    def get_preference(self, identifier, resolutions, candidates, information):
+    def get_preference(self, identifier, resolutions, candidates, information, backtrack_causes):
         return sum(1 for _ in candidates[identifier])
 
     def find_matches(self, identifier, requirements, incompatibilities):

--- a/setup.py
+++ b/setup.py
@@ -33,7 +33,7 @@ setup(
         # TODO: Remove this once 3.7 is our minimally supported version.
         "dataclasses>=0.6",
         "progress>=1.6",
-        "resolvelib>=0.7.1",
+        "resolvelib>=0.8.0",
         "html5lib>=1.1",
         "virtualenv>=20.8.1",
         "virtualenv-api>=2.1.18",

